test/server/cards/03_TWI/upgrades/AhsokasPadawanLightsaber.spec.ts

```diff
@@ -0,0 +1,67 @@
+describe('Ahsoka\'s Padawan Lightsaber', function() {
+    integration(function(contextRef) {
+        describe('Ahsoka\'s Padawan Lightsaber\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['ahsokas-padawan-lightsaber'],
+                        groundArena: ['battlefield-marine', 'ahsoka-tano#always-ready-for-trouble'],
+                        spaceArena: ['green-squadron-awing']
+                    },
+                    player2: {
+                        groundArena: ['r2d2#ignoring-protocol']
+                    }
+                });
+            });
+
+            it('initiates an attack when played on Ahsoka Tano', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.ahsokasPadawanLightsaber);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.ahsokaTanoAlwaysReadyForTrouble, context.r2d2IgnoringProtocol]);
```

Comment:
If you want you can omit the subtitle and just use the title for the unit object - e.g. `context.ahsokaTano`. Unless there are two cards in play with the same title and different subtitles, in which case you have to specify subtitle

Reply:
gotcha, that is helpful, thanks